src/components/views/elements/AccessibleButton.tsx

```diff
@@ -69,7 +69,7 @@ type IProps<T extends keyof JSX.IntrinsicElements> = DynamicHtmlElementProps<T>
     disabled?: boolean;
     className?: string;
     triggerOnMouseDown?: boolean;
-    onClick(e?: ButtonEvent): void | Promise<void>;
+    onClick?(e?: ButtonEvent): void | Promise<void>;
```

Comment:
I think it's useful that today the types will nag you to add a callback, what's the reasoning for making this optional?

Reply:
I added cleaner types for the accessiblebutton, allowing you to use it as if it was an <a href="">, with a URL.

But your reasoning is actually good, and you’re right, requiring devs to explicitly say "I need no onClick handler" is better dev experience.